### PR TITLE
Return all test results for a test run

### LINKED ISSUE #258: Add option to return all test results for a test run
[GetTestRuns](https://github.com/cake-contrib/Cake.AzureDevOps/blob/b44c19c80b165dfd9e048efb1ec92b8fb4ce8334/src/Cake.AzureDevOps/Pipelines/AzureDevOpsBuild.cs#L419) returns a list of all tets runs, and each test run contains a list of tests.
We should add a parameter to configure how many test results should be returned, including the option to return all results instead of 1000.

### PR DESCRIPTION
**Pull request by christianbumann**
Resolves #258

### CODE REVIEW

**Comment by christianbumann** [comment]
@pascalberger Maybe as you wrote, this should be a hotfix in 4.7

File: src/Cake.AzureDevOps/Pipelines/AzureDevOpsBuild.cs [hunk 1]
```diff
@@ -451,15 +451,33 @@ public IEnumerable<AzureDevOpsTestRun> GetTestRuns()
...
+            var testResults = new List<AzureDevOpsTestResult>();
+
+            while (true)
+            {
+                var results =
+                    (await testClient.GetTestResultsAsync(
+                        this.ProjectId,
+                        runId,
+                        skip: testResults.Count,
+                        top: 10000).ConfigureAwait(false))
```
Comment (christianbumann) [inline src/Cake.AzureDevOps/Pipelines/AzureDevOpsBuild.cs:463]: 10000 is the max value for top

**Comment by pascalberger** [comment]
I don't like the idea to loop through all test cases. A better, and more scalable approach would be to allow passing in filter criteria. I will update the PR accordingly.

**Review by christianbumann — COMMENTED** [review]
Your changes looks good 4 me

**Comment by pascalberger** [comment]
@christianbumann your changes have been merged, thanks for your contribution 👍

**Pull request merged**